scripts/build/collect_symbols.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+set -ex
+set -o pipefail
+
+function find_rust_platform() {
+  rustc -Vv | grep -E "^host" | perl -p -e 's/^host:\s*(\S*)/\1/'
+}
+
+PLATFORM="$(find_rust_platform)"
+
+while getopts "t:" option; do
+  case ${option} in
+  t)
+    PLATFORM="${OPTARG}"
+    ;;
+  *)
+    echo Invalid argument: -"${?}"
+    usage
+    exit 1
+    ;;
+  esac
+done
+
+if [[ -z "${PLATFORM}" ]]; then
+  echo "MUST SPECIFY PLATFORM"
+  exit 1
+fi
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+AWS_LC_DIR="${REPO_ROOT}/aws-lc-sys/aws-lc"
+SYMBOLS_FILE="${REPO_ROOT}/aws-lc-sys/symbols/${PLATFORM}.txt"
+
+
+function filter_symbols() {
+    grep -v "^_\?bignum_" | grep -v "_\?curve25519_x25519" | grep -v "_\?edwards25519_"
+}
+
+function find_libcryptos() {
+  find "${REPO_ROOT}/target" -type f -name "libcrypto.a" | grep "aws-lc-sys"
+}
+
+function find_libssls() {
+  find "${REPO_ROOT}/target" -type f -name "libssl.a" | grep "aws-lc-sys"
+}
```

Comment:
Since this is doing a find on the whole target folder can we actually make sure we only find a single `libcrypto.a` and `libssl.a`? There should only be a single one if the script is run after doing a `cargo clean && cargo build`, but if a clean is missed this could find multiple. If there are somehow multiple then we could use the wrong one potentially.

Reply:
Resolved in latest commit.